internal/reconcilers/repository.go

```diff
@@ -56,40 +58,71 @@ func (r *Reconciler) handleRepoReconcilerEvent(msg *message.Message) error {
 	}
 
 	ctx := msg.Context()
-	log.Printf("handling reconciler event for project %s and repository %d", evt.Project.String(), evt.Repository)
+	log.Printf("handling reconciler event for project %s and repository %s", evt.Project.String(), evt.EntityID.String())
 	return r.handleArtifactsReconcilerEvent(ctx, &evt)
 }
 
 // HandleArtifactsReconcilerEvent recreates the artifacts belonging to
 // an specific repository
 // nolint: gocyclo
 func (r *Reconciler) handleArtifactsReconcilerEvent(ctx context.Context, evt *messages.RepoReconcilerEvent) error {
+	entRefresh := entityMessage.NewEntityRefreshAndDoMessage().
+		WithEntityID(evt.EntityID)
+
+	m := message.NewMessage(uuid.New().String(), nil)
+	if err := entRefresh.ToMessage(m); err != nil {
+		zerolog.Ctx(ctx).Error().Err(err).Msg("error marshalling message")
+		// no point in retrying, so we return nil
+		return nil
+	}
+
+	if evt.EntityID == uuid.Nil {
+		// this might happen if we process old messages during an upgrade, but there's no point in retrying
+		zerolog.Ctx(ctx).Error().Msg("entityID is nil")
+		return nil
+	}
+
+	m.SetContext(ctx)
+	if err := r.evt.Publish(events.TopicQueueRefreshEntityByIDAndEvaluate, m); err != nil {
+		// we retry in case watermill is having a bad day
+		return fmt.Errorf("error publishing message: %w", err)
+	}
+
+	// the code below this line will be refactored in a follow-up PR
+	ewp, err := r.propService.EntityWithPropertiesByID(ctx, evt.EntityID, nil)
+	if err != nil {
+		// database might be down, so we retry
+		log.Printf("error retrieving entity with properties: %v", err)
+		return nil
+	}
+
+	repoID, err := ewp.Properties.GetProperty(properties.PropertyUpstreamID).AsInt64()
+	if err != nil {
+		log.Printf("error getting property upstreamID: %v", err)
+		// no point in retrying, so we just return nil
+		return nil
+	}
+
 	// first retrieve data for the repository
-	repository, err := r.store.GetRepositoryByRepoID(ctx, evt.Repository)
+	repository, err := r.store.GetRepositoryByRepoID(ctx, repoID)
 	if errors.Is(err, sql.ErrNoRows) {
 		zerolog.Ctx(ctx).Debug().Err(err).
-			Int64("repositoryUpstreamID", evt.Repository).
+			Int64("repositoryUpstreamID", repoID).
```

Comment:
Can we use `any` or something of the sort to print this?

Reply:
This part of the code is past the repository reconcile and now the code is trying to reconcile artifacts which are supported with github only.

Above is this code:
```
	repoID, err := ewp.Properties.GetProperty(properties.PropertyUpstreamID).AsInt64()
	if err != nil {
		log.Printf("error getting property upstreamID: %v", err)
		// no point in retrying, so we just return nil
		return nil
	}
```
so in the case you called out where the upstream ID is a string we'd hit the error case and shortcut.

I can add a unit test with upstream ID as string and split this long function to repo and artifact parts, would it help?